plasmapy/diagnostics/charged_particle_radiography.py

```diff
@@ -1419,3 +1422,249 @@ def synthetic_radiograph(
+            MeV cm\ :sup:`^2` / g, so this variable has units of MeV/cm.
+
+        active : `bool`, optional
+            If `True`, this layer is marked as an active layer. The default is `True`.
+
+        name : `str`, optional
+            An optional name for the layer.
+
+        """
+        self.thickness = thickness
+        self.energy_axis = energy_axis
+        self.stopping_power = stopping_power
+        self.active = active
+        self.name = name
+
+
+class Stack:
+    r"""
+    An ordered list of
+    `~plasmapy.diagnostics.charged_particle_radiography.Layer` objects.
+
+
+    Parameters
+    ----------
+
+    layers : list of `~plasmapy.diagnostics.charged_particle_radiography.Layer` objects
+        A list of the `~plasmapy.diagnostics.charged_particle_radiography.Layer` objects that make up the film stack.
+    """
+
+    def __init__(self, layers):
+        self._layers = layers
+        self._energy_bands = None
+
+    @property
+    def nlayers(self):
+        r"""
+        The number of layers in the stack.
+        """
+
+        return len(self._layers)
+
+    @property
+    def nactive(self):
+        r"""
+        The number of layers in the stack marked 'active'
+        """
+
+        return len([layer for layer in self._layers if layer.active])
+
+    @property
+    def thickness(self):
+        r"""
+        The total thickness of the stack.
+        """
+        thickness = np.array([layer.thickness.to(u.mm).value for layer in self._layers])
+        return np.sum(thickness) * u.mm
+
+    def deposition_curves(self, energies: u.MeV, dx=1 * u.um, return_only_active=True):
+        """
+        Calculates the deposition of an ensemble of particles over a range of
+        energies in a stack of films and filters.
+
+        Parameters
+        ----------
+
+        stack : list of `~plasmapy.diagnostics.charged_particle_radiography.Layer` objects
+            This list of `~plasmapy.diagnostics.charged_particle_radiography.Layer`
+            objects defines the composition of the film stack.
+
+        energies : `~astropy.units.Quantity` array, shape [nenergies,]
+            Energies axis over which to calculate the deposition. Units convertible
+            to eV.
+
+        dx : `~astropy.units.Quantity`, optional
+            The spatial resolution of the numerical integration of the stopping power.
+            Defaults to 1 um.
+
+        return_only_active : `bool`, optional
+            If `True`, only the deposition in layers in which the active property
+            is `True` will be returned. This is usually desirable, since particles
+            captured in other layers will not be measured. If `False`, deposition in
+            all layers of the stack are returned. The default is `True`.
+
+        Returns
+        -------
+
+        deposited : `~numpy.ndarray`, shape [nlayers, nenergies]
+            The fraction of an ensemble of each energy that will be deposited in
```

Comment:
Fraction of an ensemble of each energy? I'm just going by the docs for now, and it's not quite obvious to me what this means.

Reply:
Agreed - so what I'm trying to say is that if you start with an ensemble of N particles of energy E, this will calculate the fraction of N stopped in each layer. Then it does that for all the energies E to get a 2D array of 'deposition curves'.

I'll take a stab at rephrasing to that efffect.